### ISSUE TITLE
elm-format: I couldn't figure out what version of Elm you are using!

### ISSUE THREAD

**Issue by NeverwinterMoon** [klazuka/intellij-elm#396]
I have configured locations for Elm compiler and elm-format, but every time elm-format runs, I get the following error:

```
I couldn't figure out what version of Elm you are using!

You should either run elm-format from the directory containing
your elm.json (for Elm 0.19) or elm-package.json (for Elm 0.18),
or tell me which version of Elm with --elm-version=0.19 or --elm-version=0.18
```

Formatting works correctly nonetheless, but this output is extremely annoying, it appears every time the elm-format is used. Is it possible that the plugin would provide the suggested parameter to elm-format (--elm-version=0.19 or --elm-version=0.18) based on auto-detected version of Elm Compiler?

**Comment by klazuka** [comment]
Correct, the plugin will detect which version of Elm should be used based on the current source file. Have you attached your `elm.json` file inside IntelliJ? If so, is the file you are editing contained in a source directory of that `elm.json` file?

If none of that helps, please post your `elm.json` file.

**Comment by NeverwinterMoon** [comment]
I am using Elm 0.18, so it's still `elm-package.json` and I did specify its location to the plugin, but my project structure might be what's causing the issue then.

Here is the contents of my `elm-package.json`:

```
{
    "source-directories": [
        "src/main/elm"
    ],
    "exposed-modules": [],
    "dependencies": {
        "NoRedInk/elm-decode-pipeline": "3.0.0 <= v <= 4.0.0",
        "elm-community/random-extra": "2.0.0 <= v < 3.0.0",
        "elm-lang/core": "5.1.1 <= v < 6.0.0",
        "elm-lang/dom": "1.1.1 <= v < 2.0.0",
        "elm-lang/geolocation": "1.0.2 <= v < 2.0.0",
        "elm-lang/html": "2.0.0 <= v < 3.0.0",
        "elm-lang/http": "1.0.0 <= v < 2.0.0",
        "elm-lang/navigation": "2.1.0 <= v < 3.0.0",
        "elm-lang/virtual-dom": "2.0.4 <= v < 3.0.0",
        "evancz/url-parser": "2.0.1 <= v < 3.0.0",
        "krisajenkins/remotedata": "4.3.0 <= v < 5.0.0",
        "rtfeldman/elm-css": "9.1.0 <= v < 10.0.0",
        "rtfeldman/elm-css-helpers": "2.1.0 <= v < 3.0.0"
    },
    "elm-version": "0.18.0 <= v < 0.19.0"
}
```

As you can see, I specify the location of the sources at `source-directories`, so my `elm-package.json` is not in the same directory as the `src` containing Elm sources. Is the plugin ignoring the `source-directories` property then?

**Comment by klazuka** [comment]
What version of intellij-elm are you running?

The `source-directories` field in your `elm-package.json` *should* be honored.

When you open a file nested under `src/main/elm`, is there any kind of warning banner at the top of the file editor? If so, that would indicate that it can't find the corresponding `elm-package.json` for the current file.

**Comment by NeverwinterMoon** [comment]
Plugin version is `3.1.1`. When I open, say, src/main/elm/Main.elm, there are not warnings and everything else works as expected - navigation to imports, etc.

**Comment by klazuka** [comment]
Maybe it's timing-out running elm-format. Plugins for VSCode and others have had problems with abnormally slow start times for elm-format (maybe due to how these Haskell-built binaries are wrapped for nodejs).

Are you using nvm?

If using npm or nvm, try specifying the path to the real, native binary, not the nodejs wrapper script.

**Comment by dpinn** [comment]
I had success by specifying the fully-qualified path to the executable.

```C:\Users\david\AppData\Roaming\npm\node_modules\elm-format\unpacked_bin\elm-format.exe```

In the IntelliJ settings, the elm-format version is reported as 0.8.1, but it seems to work alright with my 0.19 project.

**Comment by NeverwinterMoon** [comment]
Oh wow, I was a prisoner of my own negligence.

First of all, I did use `npm` and the node wrapper for `elm-format`. After your advice, I did change the path to use the executable directly, not the node wrapper, but then I noticed that the output warning still had the path for `elm-format` used before that change...

Even though I had a clean install of everything, including IDEA, it seems that I've imported some of the IDEA settings from long ago without paying too much attention. One of the settings was that of a file watcher configuration I've used to run `elm-format` two years ago. So the warning was coming from that. I've removed the watcher, reverted `intellij-elm` configuration to use the node wrapper and everything worked as a charm!

Thanks for your time and sorry for the confusion!
